program/src/processor.rs

```diff
@@ -1223,6 +1381,52 @@ impl Processor {
         Ok(())
     }
 
+    #[inline(never)]
+    fn close_spot_open_orders(program_id: &Pubkey, accounts: &[AccountInfo]) -> MangoResult<()> {
+        const NUM_FIXED: usize = 7;
+        let accounts = array_ref![accounts, 0, NUM_FIXED];
+
+        let [
+            mango_group_ai,     // read
+            mango_account_ai,   // write
+            owner_ai,           // write, signer
+            dex_prog_ai,        // read
+            open_orders_ai,     // write
+            spot_market_ai,     // read
+            signer_ai,          // read
```

Comment:
I see no checks on open_orders_ai and signer_ai, is that correct?

Reply:
maybe check that `mango_account.spot_open_orders[market_index] == open_orders_ai.key`, even if it's probably covered by the CPI